src/agentscope/service/multi_modality/dash_scope.py

```diff
@@ -0,0 +1,229 @@
+        value is a list of the paths to the generated images.
+    Example:
+        prompt = "A beautiful sunset in the mountains"
+        print( dashscope_text_to_image(prompt)) gives:
+        {'status': 'SUCCESS',
+        'content': {'fig_paths':
+            ['FIG_RELATEIVE_PATH1',
+            'FIG_RELATEIVE_PATH2']}}
+    """
+
+    dashscope.api_key = api_key
+    res = dashscope.ImageSynthesis.call(
+        model=model,
+        prompt=prompt,
+        n=number_of_images,
+        size=size,
+    )
+    if res.status_code == HTTPStatus.OK:
+        if res.output.task_status == "SUCCEEDED":
+            results = res.output.results
+            # write iamge to disk
+            # if saved_dir does not exist, create it
+            fig_paths = []
+            if not os.path.exists(saved_dir):
+                os.makedirs(saved_dir)
+            for i, result in enumerate(results):
+                fig_path = f"{saved_dir}/{prompt}_{i}.png"
+                with open(fig_path, "wb+") as f:
+                    f.write(requests.get(result.url).content)
+                fig_paths.append(fig_path)
+            return ServiceResponse(
+                ServiceExecStatus.SUCCESS,
+                {"file_paths": fig_paths},
+            )
+        else:
+            err_msg = f"Task failed with status {res.output.task_status}"
+            err_msg += res.message
+            return ServiceResponse(
+                ServiceExecStatus.ERROR,
+                {"error": err_msg},
+            )
+    else:
+        err_msg = f"Error in calling the API: {res.status_code}"
+        err_msg += res.message
+        return ServiceResponse(
+            ServiceExecStatus.ERROR,
+            {"error": err_msg},
+        )
+
+
+def dashscope_image_to_text(
+    image_urls: Union[str, Tuple[str, ...]],
+    prompt: str,
+    api_key: str,
+    model: str = "qwen-vl-plus",
+) -> ServiceResponse:
+    """Generate text based on an image.
+
+    Args:
+        image_urls (`str`): the url of single or multiple images.
+        query_prompt (`str`): the text prompt.
+        api_key (`str`): The api key for the dashscope api.
+        model (`str`, defaults to 'qwen-vl-plus'): the model to use.
+    Returns:
+        ServiceResponse:
+        A dictionary with two variables: `status` and`content`.
+        If `status` is ServiceExecStatus.SUCCESS,
+        the `content` is the generated text.
+    Example:
+        image_url = "image.jpg"
+        query_prompt = "Describe the image"
+        print(image_to_text(image_url, query_prompt)) gives:
+        {'status': 'SUCCESS', 'content': 'A beautiful sunset in the mountains'}
+    """
+    dashscope.api_key = api_key
+    if not isinstance(image_urls, tuple):
+        image_urls = (image_urls,)
+    contents = []
+    for image_url in image_urls:
+        # check image url is local content or remote content
+        if not image_url.startswith(("http://", "https://")):
```

Comment:
Suggested to reuse the DashScopeMultimodalModelWrapper in AgentScope, which already handles the local/web url and prompt construct

Reply:
The DashScopeMultimodalModelWrapper handles it through a protected member function called _convert_url. Therefore, I choose to stay with the original version.

Reply:
Can also use Wrapper.format but it require the input to be a MessageBase. Seems a bit heavy for me.